### PR TITLE
Error notification client

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Notification Client for blob router](https://tools.hmcts.net/jira/browse/BPS-966)

### Change description ###

Adding minimal client setup according to the documentation. By default client is disabled and authentication is dummy. Following PR should introduce configuration setup for actual values to be used.

**NOTE**: there is a `po_box` as per documentation - it is required field. will gather information and address accordingly later. For now it's lift and shift as per hmcts/bulk-scan-processor

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/ErrorNotificationClientTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        new ErrorNotificationResponse("some id");
+
+    @Autowired
+    private ErrorNotificationClient client;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Test
+    public void should_return_Created_when_everything_is_ok_with_request() throws JsonProcessingException {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/ErrorNotificationClientTest.java]: should return notification id ?

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/config/ContextInitializer.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.env.Profiles;
+
+import static org.springframework.util.SocketUtils.findAvailableTcpPort;
+
+@Configuration
+public class ContextInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/config/ContextInitializer.java]: can we not use `@AutoConfigureWireMock` in order to be able to use Wiremock in your test ?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/clients/ErrorNotificationRequest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class ErrorNotificationRequest {
+
+    @JsonProperty("zip_file_name")
+    public final String zipFileName;
+
+    @JsonProperty("po_box")
+    public final String poBox;
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/clients/ErrorNotificationRequest.java]: could we please add a comment to remove this later once notification service provider makes it non mandatory ?

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/ErrorNotificationClientTest.java (hunk 1 above)
Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/ErrorNotificationClientTest.java]: let me review test names - literally just copied without any changes

**Pull request merged**